### DISCUSSION TITLE
Best way to find and fix broken links?

### DISCUSSION THREAD

**Discussion by nicbou** [craftcms/cms#11737] [Q&A] [answered]
Currently, I have a very suboptimal setup to fix broken links across the 300 pages of my website.

1. I have a template that displays all of my content on the single page.
2. I run a few lines of JavaScript to find and test every link on the page, then report the broken ones.
3. I use Craft's "Find and replace" utility to fix those links one by one, usually with an archive.org link.

Say that I have a table of old links and their desired replacement. What would be a safe and efficient way to fix all instances of those links on my Craft website?

**Comment by brandonkelly** [comment, marked as answer]
Craft doesn’t have a built-in way of updating broken links. The Find & Replace utility is probably your best bet.

As far as finding them, we use [Oh Dear](https://ohdear.app/) to keep track of broken links on [craftcms.com](https://craftcms.com) and it works great.

**Comment by leevigraham** [comment]
Checkout https://www.screamingfrog.co.uk/seo-spider/

**Comment by nicbou** [comment]
I ended up using the excellent linkchecker python package. Since the replacement process was very manual, the Find and Replace feature wss good enough, although a command line version would be useful. I saw that it would be doable thanks to Craft's thoughtful design.
